waspc/data/Generator/templates/sdk/wasp/server/env.ts

```diff
@@ -150,7 +150,10 @@ const serverEnvSchema = z.discriminatedUnion('NODE_ENV', [
 ])
 
 // PUBLIC API
-export const env = ensureEnvSchema(process.env, serverEnvSchema)
+export const env = ensureEnvSchema(
+  { NODE_ENV: serverDevSchema.shape.NODE_ENV.value, ...process.env },
+  serverEnvSchema,
+)
```

Comment:
Fixes this:

![image](https://github.com/user-attachments/assets/25f1d0a9-86a2-4c55-8dab-2f6d8a97ab8f)

@infomiho Couldn't find a better way to set a default value. Is this ok?